packages/nx-plugin/src/executors/internal/cli.ts

```diff
@@ -12,15 +13,49 @@ export function createCliCommandString(options?: {
   )}`;
 }
 
+export function formatCommandLog({
+  command,
+  args = [],
+  env,
+}: {
+  command: string;
+  args: string[];
+  env?: Record<string, string>;
+}): string {
+  const logElements: string[] = [];
+  if (env) {
+    const envVars = Object.entries(env).map(
+      ([key, value]) =>
+        `${chalk.green(key)}="${chalk.blueBright(value.replaceAll('"', ''))}"`,
+    );
+    // eslint-disable-next-line functional/immutable-data
+    logElements.push(...envVars);
+  }
+  // eslint-disable-next-line functional/immutable-data
+  logElements.push(chalk.cyan(command));
+  if (args.length > 0) {
+    // eslint-disable-next-line functional/immutable-data
+    logElements.push(chalk.dim.gray(args.join(' ')));
+  }
+  return logElements.join(' ');
+}
+
 export function createCliCommandObject(options?: {
   args?: Record<string, unknown>;
-  command?: string;
+  command?: 'autorun' | 'collect' | 'upload' | 'print-config' | string;
```

Comment:
TypeScript resolves this as a string, I'm afraid.

<img width="622" height="112" alt="image" src="https://github.com/user-attachments/assets/5afa75eb-afdb-444a-befb-5bd8dfde322c" />

But, there's a trick you can use get a loose autocomplete type, if that's what you want.

<img width="710" height="340" alt="image" src="https://github.com/user-attachments/assets/166f9d83-7c5a-4224-b9d0-02d9ac4d3027" />

Reply:
Ah!! I was thinking of this hack! Seems I forgot. Nice reminder of this.